test/e2e/nodes_test.go

```diff
@@ -1,49 +1,63 @@
 package e2e
 
 import (
-	"time"
-
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 
-	"k8s.io/apimachinery/pkg/types"
-
-	yaml "sigs.k8s.io/yaml"
-
 	nmstatev1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1"
 )
 
 var _ = Describe("Nodes", func() {
-	Context("when nodes are up", func() {
-		AfterEach(func() {
-			deleteNodeNeworkStates()
-		})
-		var (
-			timeout  = 5 * time.Second
-			interval = 1 * time.Second
-		)
+	Context("when are up", func() {
 		It("should have NodeNetworkState with currentState for each node", func() {
 			for _, node := range nodes {
-				key := types.NamespacedName{Namespace: namespace, Name: node}
 				var currentStateYaml nmstatev1.State
-				Eventually(func() nmstatev1.State {
-					currentStateYaml = nodeNetworkState(key).Status.CurrentState
-					return currentStateYaml
-				}, timeout, interval).ShouldNot(BeEmpty(), "Node %s should have currentState", node)
+				currentState(namespace, node, &currentStateYaml).ShouldNot(BeEmpty())
 
-				By("unmarshal state yaml into unstructured golang")
-				var currentState map[string]interface{}
-				err := yaml.Unmarshal(currentStateYaml, &currentState)
-				Expect(err).ToNot(HaveOccurred(), "Should parse correctly yaml: %s", currentStateYaml)
-
-				interfaces := currentState["interfaces"].([]interface{})
+				interfaces := interfaces(currentStateYaml)
 				Expect(interfaces).ToNot(BeEmpty(), "Node %s should have network interfaces", node)
 
 				obtainedInterfaces := interfacesName(interfaces)
-				Expect(obtainedInterfaces).To(SatisfyAll(
-					ContainElement("eth0"),
-				))
+				Expect(obtainedInterfaces).To(ContainElement("eth0"))
 			}
 		})
+		Context("and node network state is deleted", func() {
+			BeforeEach(func() {
+				deleteNodeNeworkStates()
+			})
+			It("should recreate it with currentState", func() {
+				for _, node := range nodes {
+					var currentStateYaml nmstatev1.State
+					currentState(namespace, node, &currentStateYaml).ShouldNot(BeEmpty())
```

Comment:
could you do the ShouldNot check inside currentState and just take return value with State here?

Reply:
The idea is to be the test the one that put the assert, we may want to do currentState(..).Should(BeEmpty())

Reply:
What if you move Expect outer:

Expect(currentState).Should... ?

Reply:
What do we gain with that ?

Reply:
Ah sorry, misunderstood. Well, if you think this makes more sense, let's keep it.